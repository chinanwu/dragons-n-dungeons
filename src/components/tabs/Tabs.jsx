import PropTypes from 'prop-types';
import React, { useCallback, useState } from 'react';
import { connect } from 'react-redux';

import getThemeClassName from '../../functions/getThemeClassName';
import TabList from './TabList.jsx';
import TabPanels from './TabPanels.jsx';

import './Tabs.less';

const TabTypes = {
  Tab: 'Tab',
  TabHeader: 'TabHeader',
  TabFooter: 'TabFooter',
};

const isTabType = (child, tabType) =>
  child.type &&
  child.type.name !== undefined &&
  child.type.name.toString() === tabType;

const parseTabs = tabs => {
  let tabList = [];
  let tabPanels = [];
  let tabHeader = null;
  let tabFooter = null;

  // This allows for users to have other things within Tabs as well
  let otherContents = [];

  // If there is only a single child, 'children' becomes just a single [Object object]
  if (!Array.isArray(tabs)) {
    if (isTabType(tabs, TabTypes.Tab)) {
      tabList.push({ name: tabs.props.name, disabled: tabs.props.disabled });
      tabPanels.push(tabs.props.children);
    }
    // Don't allow this for now - Maybe a future thing? Not sure
    // else if (isTabType(tabs, TabTypes.TabHeader)) {
    //   tabHeader = tabs;
    // } else if (isTabType(tabs, TabTypes.TabFooter)) {
    //   tabFooter = tabs; }
    else {
      otherContents.push(tabs);
    }
    return { tabList, tabPanels, tabHeader, tabFooter, otherContents };
  } else {
    tabs.map(tab => {
      if (isTabType(tab, TabTypes.Tab)) {
        tabList.push({ name: tab.props.name, disabled: tab.props.disabled });
        tabPanels.push(tab.props.children);
      } else if (isTabType(tab, TabTypes.TabHeader)) {
        tabHeader = tab;
      } else if (isTabType(tab, TabTypes.TabFooter)) {
        tabFooter = tab;
      } else {
        otherContents.push(tab);
      }
    });
    return { tabList, tabPanels, tabHeader, tabFooter, otherContents };
  }
};

const getFirstActive = list => {
  for (let i = 0; i < list.length; i++) {
    if (!list[i].disabled) {
      return i;
    }
  }
};

export const Tabs = ({ defaultActive, children, theme }) => {
  if (children === undefined || children.length === 0) {
    return null;
  }

  const { tabList, tabPanels, tabHeader, tabFooter, otherContents } = parseTabs(
    children
  );

  if (tabList.length === 0) {
    return <div>{otherContents}</div>;
  }

  const active = tabList[defaultActive].disabled
    ? getFirstActive(tabList)
    : defaultActive;

  const [activeIndex, setActiveIndex] = useState(active);

  const handleKeyDownTabList = useCallback(
    event => {
      if (
        !event.shiftKey &&
        !event.ctrlKey &&
        !event.altKey &&
        !event.metaKey
      ) {
        const keyCode = event.keyCode;
        if (keyCode === 39 || keyCode === 40) {
          // right or down
          event.preventDefault();

          // This is done since React state doesn't immediately update
          setActiveIndex(activeIndex => {
            const tabListLength = tabList.length;
            let nextIndex =
              activeIndex + 1 === tabListLength ? 0 : activeIndex + 1;

            while (tabList[nextIndex].disabled) {
              nextIndex = nextIndex + 1 === tabListLength ? 0 : nextIndex + 1;
            }

            return nextIndex;
          });
        } else if (keyCode === 37 || keyCode === 38) {
          // left or up
          event.preventDefault();

          // This is done since React state doesn't immediately update
          setActiveIndex(activeIndex => {
            const tabListLength = tabList.length;
            let nextIndex =
              activeIndex - 1 === -1 ? tabListLength - 1 : activeIndex - 1;

            while (tabList[nextIndex].disabled) {
              nextIndex =
                nextIndex - 1 === -1 ? tabListLength - 1 : nextIndex - 1;
            }

            return nextIndex;
          });
        }
      }
    },
    [setActiveIndex]
  );

  const handleClickTabList = useCallback(
    event => {
      setActiveIndex(
        // Unfortunate
        parseInt(event.target.attributes.id.value.replace('tabButton__', ''))
      );
    },
    [setActiveIndex]
  );

  return (
    <div className={getThemeClassName('Tabs', theme)} role="tabs">
      <TabList
        id="tabList"
        list={tabList}
        activeIndex={activeIndex}
        onClick={handleClickTabList}
        onKeyDown={handleKeyDownTabList}
        header={tabHeader}
        footer={tabFooter}
      />
      <TabPanels id="tabPanels" panels={tabPanels} activeIndex={activeIndex} />
      {otherContents.length > 0 ? otherContents : ''}
    </div>
  );
};

Tabs.propTypes = {
  defaultActive: PropTypes.number,
  children: PropTypes.any,
  theme: PropTypes.string,
};

Tabs.defaultProps = {
  defaultActive: 0,
};

export const mapStateToProps = ({ theme }) => ({
  theme: theme.theme,
});

export default connect(mapStateToProps)(Tabs);
