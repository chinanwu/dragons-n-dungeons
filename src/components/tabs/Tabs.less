@import '../styling/colours.less';

.Tabs {
  height: 100%;
}

.Tabs--dark {
  background-color: @active-tab-background-colour-dark;
}
