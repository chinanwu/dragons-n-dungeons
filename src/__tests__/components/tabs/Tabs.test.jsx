import { shallow } from 'enzyme';
import React from 'react';
import { mapStateToProps, Tabs } from '../../../components/tabs/Tabs.jsx';

jest.unmock('../../../components/tabs/Tabs.jsx');
jest.unmock('../../../functions/getThemeClassName');

describe('Tabs component', () => {
  describe('rendering', () => {
    it('renders Tabs', () => {
      const wrapper = shallow(<Tabs theme="light" />);
      expect(wrapper.getElement()).toMatchSnapshot();
    });

    it('renders Tabs with empty children', () => {
      const wrapper = shallow(<Tabs children={[]} />);
      expect(wrapper.getElement()).toMatchSnapshot();
    });

    it('renders Tabs with non-tab child', () => {
      const wrapper = shallow(
        <Tabs>
          <p>Test</p>
        </Tabs>
      );
      expect(wrapper.getElement()).toMatchSnapshot();
    });

    it('renders Tabs with tab child', () => {
      const tab = {
        type: { name: 'Tab' },
        props: {
          name: 'Test-tab',
          disabled: false,
          children: [{ type: 'p', props: { children: 'Test 12' } }],
        },
      };
      const wrapper = shallow(<Tabs children={tab} />);
      expect(wrapper.getElement()).toMatchSnapshot();
    });

    it('renders Tabs with tab children', () => {
      const tabHeader = {
        type: { name: 'TabHeader' },
        props: {
          children: [{ type: 'p', props: { children: 'Logo' } }],
        },
      };
      const tabFooter = {
        type: { name: 'TabFooter' },
        props: {
          children: [{ type: 'p', props: { children: 'Footer' } }],
        },
      };

      const children = [
        tabHeader,
        {
          type: { name: 'Tab' },
          props: {
            name: 'Test-tab',
            disabled: true,
            children: [{ type: 'p', props: { children: 'Test 1' } }],
          },
        },
        {
          type: { name: 'Tab' },
          props: {
            name: 'Test-tab2',
            disabled: false,
            children: [
              {
                type: 'div',
                props: {
                  children: [{ type: 'p', props: { children: 'Test 2' } }],
                },
              },
            ],
          },
        },
        tabFooter,
        {
          type: 'p',
          props: { children: 'Test 3' },
        },
      ];
      const wrapper = shallow(<Tabs children={children} />);
      expect(wrapper.getElement()).toMatchSnapshot();
    });
  });

  describe('behaviour', () => {
    it('switch to tab when tab list button is clicked', () => {
      const tab = {
        type: { name: 'Tab' },
        props: {
          name: 'Test-tab',
          disabled: false,
          children: [{ type: 'p', props: { children: 'Test 1' } }],
        },
      };
      const wrapper = shallow(<Tabs children={tab} />);
      expect(wrapper.getElement()).toMatchSnapshot();

      const event = {
        target: {
          attributes: {
            id: { value: 'tabButton__0' },
          },
        },
      };
      wrapper.find('#tabList').simulate('click', event);
    });

    it('when keydown with unsupported keys, do not prevent default', () => {
      const preventDefault = jest.fn();
      const shiftEvent = { shiftKey: true, preventDefault: preventDefault };
      const ctrlEvent = { ctrlKey: true, preventDefault: preventDefault };
      const altEvent = { altKey: true, preventDefault: preventDefault };
      const metaEvent = { metaKey: true, preventDefault: preventDefault };
      const slashEvent = { keyCode: 191, preventDefault: preventDefault };

      const tab = {
        type: { name: 'Tab' },
        props: {
          name: 'Test-tab',
          disabled: false,
          children: [{ type: 'p', props: { children: 'Test 1' } }],
        },
      };

      const wrapper = shallow(<Tabs children={tab} />);
      expect(wrapper.getElement()).toMatchSnapshot();

      wrapper.find('#tabList').simulate('keydown', shiftEvent);
      wrapper.find('#tabList').simulate('keydown', ctrlEvent);
      wrapper.find('#tabList').simulate('keydown', altEvent);
      wrapper.find('#tabList').simulate('keydown', metaEvent);
      wrapper.find('#tabList').simulate('keydown', slashEvent);
      expect(preventDefault).not.toHaveBeenCalled();
    });

    it('when left is pressed, go to previous tab', () => {
      const tabs = [
        {
          type: { name: 'Tab' },
          props: {
            name: 'Test-tab',
            disabled: false,
            children: [{ type: 'p', props: { children: 'Test 1' } }],
          },
        },
        {
          type: { name: 'Tab' },
          props: {
            name: 'Test-tab2',
            disabled: false,
            children: [{ type: 'p', props: { children: 'Test 2' } }],
          },
        },
        {
          type: { name: 'Tab' },
          props: {
            name: 'Test-tab3',
            disabled: true,
            children: [{ type: 'p', props: { children: 'Test 3' } }],
          },
        },
      ];
      const wrapper = shallow(<Tabs defaultActive={0} children={tabs} />);
      expect(wrapper.getElement()).toMatchSnapshot();

      const preventDefault = jest.fn();
      const event = {
        keyCode: 37,
        preventDefault: preventDefault,
      };

      wrapper.find('#tabList').simulate('keydown', event);
      expect(preventDefault).toHaveBeenCalledTimes(1);
    });

    it('when left is pressed and previous tab is disabled, go to the first previous non-disabled tab', () => {
      const tabs = [
        {
          type: { name: 'Tab' },
          props: {
            name: 'Test-tab',
            disabled: true,
            children: [{ type: 'p', props: { children: 'Test 1' } }],
          },
        },
        {
          type: { name: 'Tab' },
          props: {
            name: 'Test-tab2',
            disabled: false,
            children: [{ type: 'p', props: { children: 'Test 2' } }],
          },
        },
        {
          type: { name: 'Tab' },
          props: {
            name: 'Test-tab3',
            disabled: false,
            children: [{ type: 'p', props: { children: 'Test 3' } }],
          },
        },
      ];

      const wrapper = shallow(<Tabs defaultActive={1} children={tabs} />);
      expect(wrapper.getElement()).toMatchSnapshot();

      const preventDefault = jest.fn();
      const event = {
        keyCode: 37,
        preventDefault: preventDefault,
      };

      wrapper.find('#tabList').simulate('keydown', event);
      expect(preventDefault).toHaveBeenCalledTimes(1);
    });

    it('when right or down arrow is pressed, go to the tab to the right', () => {
      const tabs = [
        {
          type: { name: 'Tab' },
          props: {
            name: 'Test-tab',
            disabled: true,
            children: [{ type: 'p', props: { children: 'Test 1' } }],
          },
        },
        {
          type: { name: 'Tab' },
          props: {
            name: 'Test-tab2',
            disabled: false,
            children: [{ type: 'p', props: { children: 'Test 2' } }],
          },
        },
        {
          type: { name: 'Tab' },
          props: {
            name: 'Test-tab3',
            disabled: false,
            children: [{ type: 'p', props: { children: 'Test 3' } }],
          },
        },
      ];
      const wrapper = shallow(<Tabs defaultActive={2} children={tabs} />);
      expect(wrapper.getElement()).toMatchSnapshot();

      const preventDefault = jest.fn();
      const event = {
        keyCode: 39,
        preventDefault: preventDefault,
      };
      wrapper.find('#tabList').simulate('keydown', event);
      expect(preventDefault).toHaveBeenCalledTimes(1);
    });

    it('when right is pressed, and next is disabled, go to the next non-disabled tab', () => {
      const tabs = [
        {
          type: { name: 'Tab' },
          props: {
            name: 'Test-tab',
            disabled: false,
            children: [{ type: 'p', props: { children: 'Test 1' } }],
          },
        },
        {
          type: { name: 'Tab' },
          props: {
            name: 'Test-tab2',
            disabled: false,
            children: [{ type: 'p', props: { children: 'Test 2' } }],
          },
        },
        {
          type: { name: 'Tab' },
          props: {
            name: 'Test-tab3',
            disabled: true,
            children: [{ type: 'p', props: { children: 'Test 3' } }],
          },
        },
      ];
      const wrapper = shallow(<Tabs defaultActive={1} children={tabs} />);
      expect(wrapper.getElement()).toMatchSnapshot();

      const preventDefault = jest.fn();
      const event = {
        keyCode: 39,
        preventDefault: preventDefault,
      };

      wrapper.find('#tabList').simulate('keydown', event);
      expect(preventDefault).toHaveBeenCalledTimes(1);
    });

    it('when home is pressed, go to first active tab', () => {
      const tabs = [
        {
          type: { name: 'Tab' },
          props: {
            name: 'Test-tab',
            disabled: true,
            children: [{ type: 'p', props: { children: 'Test 1' } }],
          },
        },
        {
          type: { name: 'Tab' },
          props: {
            name: 'Test-tab2',
            disabled: false,
            children: [{ type: 'p', props: { children: 'Test 2' } }],
          },
        },
        {
          type: { name: 'Tab' },
          props: {
            name: 'Test-tab3',
            disabled: false,
            children: [{ type: 'p', props: { children: 'Test 3' } }],
          },
        },
      ];
      const wrapper = shallow(<Tabs defaultActive={0} children={tabs} />);
      expect(wrapper.getElement()).toMatchSnapshot();

      const preventDefault = jest.fn();
      const event = {
        keyCode: 36,
        preventDefault: preventDefault,
      };

      wrapper.find('#tabList').simulate('keydown', event);
      expect(preventDefault).toHaveBeenCalledTimes(1);
    });

    it('when end is pressed, go to last active tab', () => {
      const tabs = [
        {
          type: { name: 'Tab' },
          props: {
            name: 'Test-tab',
            disabled: false,
            children: [{ type: 'p', props: { children: 'Test 1' } }],
          },
        },
        {
          type: { name: 'Tab' },
          props: {
            name: 'Test-tab2',
            disabled: false,
            children: [{ type: 'p', props: { children: 'Test 2' } }],
          },
        },
        {
          type: { name: 'Tab' },
          props: {
            name: 'Test-tab3',
            disabled: true,
            children: [{ type: 'p', props: { children: 'Test 3' } }],
          },
        },
      ];
      const wrapper = shallow(<Tabs defaultActive={0} children={tabs} />);
      expect(wrapper.getElement()).toMatchSnapshot();

      const preventDefault = jest.fn();
      const event = {
        keyCode: 35,
        preventDefault: preventDefault,
      };

      wrapper.find('#tabList').simulate('keydown', event);
      expect(preventDefault).toHaveBeenCalledTimes(1);
    });
  });

  describe('mapStateToProps', () => {
    it('returned mapped properties', () => {
      const theme = 'light';
      const state = { theme };
      expect(mapStateToProps({ theme: state })).toEqual({ theme: theme });
    });
  });
});
