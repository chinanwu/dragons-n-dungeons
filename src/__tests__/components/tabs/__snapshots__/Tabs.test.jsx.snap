// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs component behaviour switch to tab when tab list button is clicked 1`] = `
<div
  className="Tabs"
  role="tabs"
>
  <Memo(Connect(TabList))
    activeIndex={0}
    footer={null}
    header={null}
    id="tabList"
    list={
      Array [
        Object {
          "disabled": false,
          "name": "Test-tab",
        },
      ]
    }
    onClick={[Function]}
    onKeyDown={[Function]}
  />
  <Memo(Connect(TabPanels))
    activeIndex={0}
    id="tabPanels"
    panels={
      Array [
        Array [
          Object {
            "props": Object {
              "children": "Test 1",
            },
            "type": "p",
          },
        ],
      ]
    }
  />
  
</div>
`;

exports[`Tabs component behaviour when keydown with unsupported keys, do not prevent default 1`] = `
<div
  className="Tabs"
  role="tabs"
>
  <Memo(Connect(TabList))
    activeIndex={0}
    footer={null}
    header={null}
    id="tabList"
    list={
      Array [
        Object {
          "disabled": false,
          "name": "Test-tab",
        },
      ]
    }
    onClick={[Function]}
    onKeyDown={[Function]}
  />
  <Memo(Connect(TabPanels))
    activeIndex={0}
    id="tabPanels"
    panels={
      Array [
        Array [
          Object {
            "props": Object {
              "children": "Test 1",
            },
            "type": "p",
          },
        ],
      ]
    }
  />
  
</div>
`;

exports[`Tabs component behaviour when left is pressed and next tab is disabled, go to the next non-disabled tab 1`] = `
<div
  className="Tabs"
  role="tabs"
>
  <Memo(Connect(TabList))
    activeIndex={1}
    footer={null}
    header={null}
    id="tabList"
    list={
      Array [
        Object {
          "disabled": true,
          "name": "Test-tab",
        },
        Object {
          "disabled": false,
          "name": "Test-tab2",
        },
        Object {
          "disabled": false,
          "name": "Test-tab3",
        },
      ]
    }
    onClick={[Function]}
    onKeyDown={[Function]}
  />
  <Memo(Connect(TabPanels))
    activeIndex={1}
    id="tabPanels"
    panels={
      Array [
        Array [
          Object {
            "props": Object {
              "children": "Test 1",
            },
            "type": "p",
          },
        ],
        Array [
          Object {
            "props": Object {
              "children": "Test 2",
            },
            "type": "p",
          },
        ],
        Array [
          Object {
            "props": Object {
              "children": "Test 3",
            },
            "type": "p",
          },
        ],
      ]
    }
  />
  
</div>
`;

exports[`Tabs component behaviour when left is pressed and previous tab is disabled, go to the first previous non-disabled tab 1`] = `
<div
  className="Tabs"
  role="tabs"
>
  <Memo(Connect(TabList))
    activeIndex={1}
    footer={null}
    header={null}
    id="tabList"
    list={
      Array [
        Object {
          "disabled": true,
          "name": "Test-tab",
        },
        Object {
          "disabled": false,
          "name": "Test-tab2",
        },
        Object {
          "disabled": false,
          "name": "Test-tab3",
        },
      ]
    }
    onClick={[Function]}
    onKeyDown={[Function]}
  />
  <Memo(Connect(TabPanels))
    activeIndex={1}
    id="tabPanels"
    panels={
      Array [
        Array [
          Object {
            "props": Object {
              "children": "Test 1",
            },
            "type": "p",
          },
        ],
        Array [
          Object {
            "props": Object {
              "children": "Test 2",
            },
            "type": "p",
          },
        ],
        Array [
          Object {
            "props": Object {
              "children": "Test 3",
            },
            "type": "p",
          },
        ],
      ]
    }
  />
  
</div>
`;

exports[`Tabs component behaviour when left is pressed, go to previous tab 1`] = `
<div
  className="Tabs"
  role="tabs"
>
  <Memo(Connect(TabList))
    activeIndex={0}
    footer={null}
    header={null}
    id="tabList"
    list={
      Array [
        Object {
          "disabled": false,
          "name": "Test-tab",
        },
        Object {
          "disabled": false,
          "name": "Test-tab2",
        },
        Object {
          "disabled": true,
          "name": "Test-tab3",
        },
      ]
    }
    onClick={[Function]}
    onKeyDown={[Function]}
  />
  <Memo(Connect(TabPanels))
    activeIndex={0}
    id="tabPanels"
    panels={
      Array [
        Array [
          Object {
            "props": Object {
              "children": "Test 1",
            },
            "type": "p",
          },
        ],
        Array [
          Object {
            "props": Object {
              "children": "Test 2",
            },
            "type": "p",
          },
        ],
        Array [
          Object {
            "props": Object {
              "children": "Test 3",
            },
            "type": "p",
          },
        ],
      ]
    }
  />
  
</div>
`;

exports[`Tabs component behaviour when left or up arrow is pressed, go to the tab to the left 1`] = `
<div
  className="Tabs"
  role="tabs"
>
  <Memo(Connect(TabList))
    activeIndex={0}
    footer={null}
    header={null}
    id="tabList"
    list={
      Array [
        Object {
          "disabled": false,
          "name": "Test-tab",
        },
        Object {
          "disabled": false,
          "name": "Test-tab2",
        },
        Object {
          "disabled": true,
          "name": "Test-tab3",
        },
      ]
    }
    onClick={[Function]}
    onKeyDown={[Function]}
  />
  <Memo(Connect(TabPanels))
    activeIndex={0}
    id="tabPanels"
    panels={
      Array [
        Array [
          Object {
            "props": Object {
              "children": "Test 1",
            },
            "type": "p",
          },
        ],
        Array [
          Object {
            "props": Object {
              "children": "Test 2",
            },
            "type": "p",
          },
        ],
        Array [
          Object {
            "props": Object {
              "children": "Test 3",
            },
            "type": "p",
          },
        ],
      ]
    }
  />
  
</div>
`;

exports[`Tabs component behaviour when right is pressed, and next is disabled, go to the next non-disabled tab 1`] = `
<div
  className="Tabs"
  role="tabs"
>
  <Memo(Connect(TabList))
    activeIndex={1}
    footer={null}
    header={null}
    id="tabList"
    list={
      Array [
        Object {
          "disabled": false,
          "name": "Test-tab",
        },
        Object {
          "disabled": false,
          "name": "Test-tab2",
        },
        Object {
          "disabled": true,
          "name": "Test-tab3",
        },
      ]
    }
    onClick={[Function]}
    onKeyDown={[Function]}
  />
  <Memo(Connect(TabPanels))
    activeIndex={1}
    id="tabPanels"
    panels={
      Array [
        Array [
          Object {
            "props": Object {
              "children": "Test 1",
            },
            "type": "p",
          },
        ],
        Array [
          Object {
            "props": Object {
              "children": "Test 2",
            },
            "type": "p",
          },
        ],
        Array [
          Object {
            "props": Object {
              "children": "Test 3",
            },
            "type": "p",
          },
        ],
      ]
    }
  />
  
</div>
`;

exports[`Tabs component behaviour when right or down arrow is pressed, go to the tab to the right 1`] = `
<div
  className="Tabs"
  role="tabs"
>
  <Memo(Connect(TabList))
    activeIndex={2}
    footer={null}
    header={null}
    id="tabList"
    list={
      Array [
        Object {
          "disabled": true,
          "name": "Test-tab",
        },
        Object {
          "disabled": false,
          "name": "Test-tab2",
        },
        Object {
          "disabled": false,
          "name": "Test-tab3",
        },
      ]
    }
    onClick={[Function]}
    onKeyDown={[Function]}
  />
  <Memo(Connect(TabPanels))
    activeIndex={2}
    id="tabPanels"
    panels={
      Array [
        Array [
          Object {
            "props": Object {
              "children": "Test 1",
            },
            "type": "p",
          },
        ],
        Array [
          Object {
            "props": Object {
              "children": "Test 2",
            },
            "type": "p",
          },
        ],
        Array [
          Object {
            "props": Object {
              "children": "Test 3",
            },
            "type": "p",
          },
        ],
      ]
    }
  />
  
</div>
`;

exports[`Tabs component rendering renders Tabs 1`] = `null`;

exports[`Tabs component rendering renders Tabs with empty children 1`] = `null`;

exports[`Tabs component rendering renders Tabs with non-tab child 1`] = `
<div>
  <p>
    Test
  </p>
</div>
`;

exports[`Tabs component rendering renders Tabs with tab child 1`] = `
<div
  className="Tabs"
  role="tabs"
>
  <Memo(Connect(TabList))
    activeIndex={0}
    footer={null}
    header={null}
    id="tabList"
    list={
      Array [
        Object {
          "disabled": false,
          "name": "Test-tab",
        },
      ]
    }
    onClick={[Function]}
    onKeyDown={[Function]}
  />
  <Memo(Connect(TabPanels))
    activeIndex={0}
    id="tabPanels"
    panels={
      Array [
        Array [
          Object {
            "props": Object {
              "children": "Test 12",
            },
            "type": "p",
          },
        ],
      ]
    }
  />
  
</div>
`;

exports[`Tabs component rendering renders Tabs with tab children 1`] = `
<div
  className="Tabs"
  role="tabs"
>
  <Memo(Connect(TabList))
    activeIndex={1}
    footer={
      Object {
        "props": Object {
          "children": Array [
            Object {
              "props": Object {
                "children": "Footer",
              },
              "type": "p",
            },
          ],
        },
        "type": Object {
          "name": "TabFooter",
        },
      }
    }
    header={
      Object {
        "props": Object {
          "children": Array [
            Object {
              "props": Object {
                "children": "Logo",
              },
              "type": "p",
            },
          ],
        },
        "type": Object {
          "name": "TabHeader",
        },
      }
    }
    id="tabList"
    list={
      Array [
        Object {
          "disabled": true,
          "name": "Test-tab",
        },
        Object {
          "disabled": false,
          "name": "Test-tab2",
        },
      ]
    }
    onClick={[Function]}
    onKeyDown={[Function]}
  />
  <Memo(Connect(TabPanels))
    activeIndex={1}
    id="tabPanels"
    panels={
      Array [
        Array [
          Object {
            "props": Object {
              "children": "Test 1",
            },
            "type": "p",
          },
        ],
        Array [
          Object {
            "props": Object {
              "children": Array [
                Object {
                  "props": Object {
                    "children": "Test 2",
                  },
                  "type": "p",
                },
              ],
            },
            "type": "div",
          },
        ],
      ]
    }
  />
  Object {
    "props": Object {
      "children": "Test 3",
    },
    "type": "p",
  }
</div>
`;
